src/core/lib/security/credentials/external/aws_external_account_credentials.cc

```diff
@@ -73,6 +77,18 @@ std::string UrlEncode(const absl::string_view& s) {
   return result;
 }
 
+bool ValidateAwsUrl(const std::string& urlString) {
+  absl::StatusOr<grpc_core::URI> url = grpc_core::URI::Parse(urlString);
```

Comment:
You'll need to remove all of the `grpc_core::` prefixes in this code, since this function is already inside of that namespace.